sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -446,7 +446,18 @@ void CurlSession::ParseChunkSize(Context const& context)
       if (i > 1 && this->m_readBuffer[index] == '\n')
       {
         // get chunk size. Chunk size comes in Hex value
-        this->m_chunkSize = static_cast<int64_t>(std::stoull(strChunkSize, nullptr, 16));
+        try
+        {
+          this->m_chunkSize = static_cast<int64_t>(std::stoull(strChunkSize, nullptr, 16));
+        }
+        catch (std::invalid_argument& err)
+        {
+          (void)err;
+          // Server can return something like `\n\r\n` for a chunk of zero length data. This is
+          // allowed by RFC. `stoull` will throw invalid_argument if there is not at least one hex
+          // digit to be parsed. For those cases, we consider the response as zero-lenght.
+          this->m_chunkSize = 0;
+        }
```

Comment:
Don't use exception handling/error checking as control flow. Add the appropriate explicit checks up-front before calling stoull.

Reply:
why? std::stoul already does it

Reply:
Exceptions as control flow is an anti-pattern:
https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why

https://wiki.c2.com/?DontUseExceptionsForFlowControl

Reply:
Add a check, if first character in `strChunkSize` is not a digit, set chunksize to 0, else call stoull.

Also, what happens today if you have `abc`, without a `\r\n`, does strChunkSize still end up getting set to 0? Is that what we want?

Reply:
>Add a check, if first character in strChunkSize is not a digit

We can have digits from `a` to `f`.

I get it about the try is not the best, but in this case it saves a lot of extra coding that ended up duplicating the validation from stoul.  if stould would return some error code instead of throwing, then I it would be better, but in this case, stoul is meant for you to handle the error in case it happen.

>Also, what happens today if you have abc, without a \r\n
We don't call stoul until we find `\r\n`,  then, there could be a number like `33XXXX\r\n` that is pass to stoul and that's fine, it means the size is 33 and the XXXXX is just garbage that the RFC don't explicitly block.

I see more benefit on letting this case just being a special case where we rely on a well tested function (stoul) instead of doing it manually and required to add tests for all possible things that can came as input

Reply:
I understand. How about something simple, like this. For any other exception cases from call to stoull, let them bubble up to the caller/don't catch.
```C++
if (isxdigit(strChunkSize[0]))
{
   this->m_chunkSize = static_cast<int64_t>(std::stoull(strChunkSize, nullptr, 16));
}
else
{
   this->m_chunkSize = 0;
}
```

Reply:
We can't bubble up because this is while reading from the network.
What does it mean an invalid argument in the middle of a download. We need to convert this to a TransportException anyway

Reply:
> We need to convert this to a TransportException anyway

That's fine to do.